WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs

```diff
@@ -1052,18 +1057,39 @@ private async Task AddItemToArray()
                     innerType = innerType.GetGenericTypeDefinition();
                 }
 
-                var existing = new ObservableCollection<string>(AppDomain.CurrentDomain.GetAssemblies()
+                var types = AppDomain.CurrentDomain.GetAssemblies()
                     .SelectMany(s => s.GetTypes())
                     .Where(p => innerType.IsAssignableFrom(p) && p.IsClass && !p.IsAbstract)
-                    .Select(x => x.Name));
+                    .Select(x => new TypeEntry(x.Name, "", x))
+                    .ToList();
 
                 // no inheritable
-                if (existing.Count == 1)
+                if (types.Count == 1)
                 {
                     var type = arr.InnerType;
                     if (type == typeof(CKeyValuePair))
                     {
-                        await _appViewModel.SetActiveDialog(new SelectRedTypeDialogViewModel
+                        types = new List<TypeEntry>
+                        {
+                            new("Color", "Color description", typeof(CColor)),
+                            new("CpuNameU64", "", typeof(CName)),
+                            new("Cube", "Reference to cube xbm", typeof(CResourceReference<ITexture>)),
+                            new("DynamicTexture", "Reference to dtex file", typeof(CResourceReference<ITexture>)),
+                            new("FoliageParameters", "Reference to fp file", typeof(CResourceReference<CFoliageProfile>)),
+                            new("Gradient", "Reference to gradient file", typeof(CResourceReference<CGradient>)),
+                            new("HairParameters", "Reference to hp file", typeof(CResourceReference<CHairProfile>)),
+                            new("MultilayerMask", "Reference to mlmask file", typeof(CResourceReference<Multilayer_Mask>)),
+                            new("MultilayerSetup", "Reference to mlsetup file", typeof(CResourceReference<Multilayer_Setup>)),
+                            new("Scalar", "", typeof(CFloat)),
```

Comment:
this could be "a decimal number". Also, maybe `Integer` is missing here? Or can we use `Scalar` interchangeably and the engine will cope?